optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/factory/MoveIteratorFactory.java

```diff
@@ -49,7 +50,7 @@
      * @throws UnsupportedOperationException if only {@link #createRandomMoveIterator(ScoreDirector, Random)} is
      * supported
      */
-    Iterator<Move> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
+    Iterator<Move_> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
```

Comment:
Notice that even just with commits 1 & 2 "backward compatibility" is broken, but not in all cases. If the implementing (overriding) method returns `List<MyMove>` and `MyMove extends AbstractMove<MySolution>` then the factory source will compile. If `MyMove extends AbstractMove` without the `Solution_` parameter then it will not compile. And if it returns `List<? extends Move>`, which was the return type declared by the interface before this change, then it won't compile either.

I'm going to update the upgrade recipe with this instructions.